content/en/docs/ops/integrations/spire/automatic_registration_test.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+# shellcheck disable=SC2034,SC2153,SC2154,SC2155,SC2164
+
+# Copyright Istio Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+# Install SPIRE configured with k8s Controller Manager
+snip_install_spire_with_controller_manager
+_wait_for_daemonset spire spire-agent
+_wait_for_deployment spire spire-server
+
+# Install Istio
+# @setup profile=none
+set +u # Do not exit when value is unset. CHECK_FILE in the IstioOperator might be unset
+snip_define_istio_operator
+if ! istioctl install --set tag="$TAG" --set hub="$HUB" --skip-confirmation -f ./istio.yaml
```

Comment:
Note: Since the istio-ingressgateway will not be ready, this command will take at least 5 minutes to execute before it returns an error. I can set the readiness timeout on install to be less than 5 minutes, but I don't want this to cause the other components to fail.

Reply:
Would it be reasonable to set the readiness timeout to 30s or 1m?

Reply:
I'm not sure I understand exactly why it is expected to time out, but would it be better if you use `istioctl manifest generate | kubectl apply` instead of `istioctl install`? https://preliminary.istio.io/latest/docs/setup/install/istioctl/#generate-a-manifest-before-installation

Reply:
I was installing Istio before applying the ClusterSPIFFEID. Therefore the istio-ingressgateway wasn't registered with SPIRE and couldn't obtain its certificates. The install command would timeout after 5 min of waiting for the gateway to be ready. This was the expected behavior given how the demo was organized. To improve the customer experience and testability of the demo I have modified the demo's workflow so that the ClusterSPIFFEID is created prior to installing Istio. The Istio installation should complete successfully with this change.